pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -182,6 +187,17 @@ func (e *HookExecutor) executeExecNewPod(hook *deployapi.LifecycleHook, deployme
 waitLoop:
 	for {
 		updatedPod = nextPod()
+		if len(updatedPod.Status.ContainerStatuses) > 0 {
+			restarts := updatedPod.Status.ContainerStatuses[0].RestartCount
+			if hook.ExecNewPod.MaxRestarts > 0 && restarts >= hook.ExecNewPod.MaxRestarts {
+				glog.V(0).Infof("Maximum (%d) lifecycle pod %s/%s restarts reached, deleting pod", restarts, pod.Namespace, pod.Name)
+				updatedPod.Spec.RestartPolicy = kapi.RestartPolicyNever
```

Comment:
leftover... will delete it
On Mar 3, 2016 20:05, "David Eads" <notifications@github.com> wrote:

> In pkg/deploy/strategy/support/lifecycle.go
> <https://github.com/openshift/origin/pull/7767#discussion_r54929685>:
>
> > @@ -182,6 +187,17 @@ func (e *HookExecutor) executeExecNewPod(hook *deployapi.LifecycleHook, deployme
> >  waitLoop:
> >  	for {
> >  		updatedPod = nextPod()
> > +		if len(updatedPod.Status.ContainerStatuses) > 0 {
> > +			restarts := updatedPod.Status.ContainerStatuses[0].RestartCount
> > +			if hook.ExecNewPod.MaxRestarts > 0 && restarts >= hook.ExecNewPod.MaxRestarts {
> > +				glog.V(0).Infof("Maximum (%d) lifecycle pod %s/%s restarts reached, deleting pod", restarts, pod.Namespace, pod.Name)
> > +				updatedPod.Spec.RestartPolicy = kapi.RestartPolicyNever
>
> Why are you setting this here? You're never using the updatedPod since
> you're returning from this block, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7767/files#r54929685>.
>